cylc/flow/platforms.py

```diff
@@ -248,10 +248,13 @@ def get_platform_from_group(
     TODO: Uses host_selection methods; should also allow custom select methods.
     """
     if bad_hosts:
-        good_platforms = set()
+        good_platforms = []
         for platform in group['platforms']:
-            if set(platform_from_name(platform)['hosts']) - bad_hosts:
-                good_platforms.add(platform)
+            if [
+                host for host in platform_from_name(platform)['hosts']
+                if host not in bad_hosts
+            ]:
```

Comment:
Can do
```suggestion
            if set(platform_from_name(platform)['hosts']).difference(bad_hosts):
```
> Note, the non-operator versions of union(), intersection(), difference(), and symmetric_difference(), issubset(), and issuperset() methods will accept any iterable as an argument

https://docs.python.org/3.7/library/stdtypes.html#set

Or
```suggestion
            if any(
                host for host in platform_from_name(platform)['hosts']
                if host not in bad_hosts
            ):
```

Reply:
Can't use the first solution because conversion to set will remove any ordering.

Reply:
But surely the order doesn't matter there - all that's doing is checking if there are any hosts that are are not bad?

Reply:
Basically, as far as I can see, the only diff that's needed is
```diff
-        good_platforms = set()
+        good_platforms = []
         for platform in group['platforms']:
             if set(platform_from_name(platform)['hosts']) - bad_hosts:
-                good_platforms.add(platform)
+                good_platforms.append(platform)
```